server/controllers/wishlistController.js

```diff
@@ -0,0 +1,202 @@
+  });
+
+  return totalCost;
+};
+
+exports.addItem = (req, res) => {
+  // Bad request if no body is sent
+  if (Object.keys(req.body).length === 0) {
+    return res.status(400).json({
+      Error: 'Body missing data',
+    });
+  }
+
+  const familyId = req.params.familyId;
+  const item = req.body.item;
+
+  // TODO: Check structure of item
+  // Prevent duplicate items
+
+  try {
+    Wishlist.find({ family: familyId }).exec((err, wishlist) => {
+      if (err) {
+        res.status(500).json({ Error: err });
+      }
+
+      wishlist[0].list.push(item);
+
+      wishlist[0].totalListCost = _getWishlistCost(wishlist[0].list);
+
+      wishlist[0].save((err, wishlist) => {
+        if (err) {
+          res.status(500).json({ Error: err });
+        }
+
+        res.status(201).json({
+          message: 'Successfully added item to wishlist',
+          wishlist: wishlist,
+        });
+      });
+    });
+  } catch (err) {
+    res.status(500).json({ Message: 'TEMP ERROR TEST' });
+  }
+};
+
+exports.create = (req, res) => {
+  // Bad request if no body is sent
+  if (Object.keys(req.body).length === 0) {
+    return res.status(400).json({
+      Error: 'Body missing data',
+    });
+  }
+
+  const familyId = req.params.familyId;
+  const newWishlist = req.body.list;
+
+  // Check newWishlist object type is an Array
+  // Single items should still reside in an Array
+
+  try {
+    Wishlist.where({ family: familyId }).exec((err, wishlist) => {
+      if (err) {
+        res.status(500).json({ Error: err });
+      }
+
+      if (wishlist[0].list !== []) {
+        return res.status(200).json({
+          Message: 'Unable to create wishlist. Wishlist already exists.',
+        });
+      }
+
+      wishlist[0].list = newWishlist;
+
+      wishlist[0].totalListCost = _getWishlistCost(wishlist[0].list);
+
+      wishlist[0].save((err, wishlist) => {
+        if (err) {
+          res.status(500).json({ Error: err });
+        }
+
+        res.status(200).json({
+          message: 'Successfully added new wishlist to family.',
+          wishlist: wishlist,
+        });
+      });
+    });
+  } catch (err) {
+    res.status(500).json({ Message: 'TEMP ERROR TEST' });
+  }
+};
+
+exports.delete = (req, res) => {
+  res.send('NOT YET IMPLEMENTED');
+};
+
+exports.removeItem = (req, res) => {
+  // Bad request if no body is sent
+  if (Object.keys(req.body).length === 0) {
+    return res.status(400).json({
+      Error: 'Body missing data',
+    });
+  }
+
+  const familyId = req.params.familyId;
+  const { itemName } = req.body.item;
+  const itemId = req.body.itemId;
+
+  try {
+    Wishlist.where({ family: familyId })
+      .findOne({
+        'list.itemName': itemName,
+        'list._id': itemId,
+      })
+      .exec((err, wishlist) => {
+        if (err) {
+          res.status(500).json({ Error: err });
+        }
+
+        if (!wishlist) {
+          return res
+            .status(400)
```

Comment:
Your error states 'not found', should we set this to 404?

Reply:
Also lets set this to one line to keep it consistent with all the other responses, like `res.status(500).json({ Error: err });` (ln 124)